pkg/controller/grafanadatasource/datasource_controller.go

```diff
@@ -168,10 +180,22 @@ func (r *ReconcileGrafanaDataSource) reconcileDataSources(state *common.DataSour
 		}
 	}
 
-	// apply dataSourcesToDelete
+	// apply dataSources config maps to delete. Can be multiple data sources in CM
 	for _, ds := range dataSourcesToDelete {
-		log.Info(fmt.Sprintf("deleting datasource %v", ds))
+		log.Info(fmt.Sprintf("deleting datasource config map %v", ds))
 		if state.KnownDataSources.Data != nil {
+			dsList, err := r.fetchDataSourceNames(state.KnownDataSources, ds)
+			if err != nil {
+				log.Error(err, fmt.Sprintf("error fetching datasources name from CM %v %v", ds, err))
+				return err
+			}
+			for _, dsName := range dsList {
+				resp, err := grafanaClient.DeleteDataSourceByName(dsName)
```

Comment:
It looks like the technique you've chosen here is to workaround a grafana bug by making the request directly to the grafana api. Did you investigate why this is needed? Since the logic in the rest of this section builds the grafana config file, I would have expected to see the datasource appended to the [`deleteDatasources` section in the config](https://grafana.com/docs/grafana/latest/administration/provisioning/#data-sources) rather than hitting the API.

Reply:
Yes, I've chosen to use API call to delete data sources from Grafana. I've considered an option to removing DS though `deleteDatasources`, I've mentioned this in the corresponding ticket. But, I decided that this method is not convenient in this case because:
-  we'll need to find if a newly added data source exists in `deleteDatasource` section and remove it.
-  adding `deleteDatasources` section will increase the data source config map size which is limited by 1 Mb.
- `deleteDatasources` section won't be linked with any CR.
So, for those reasons I decided to use API call to remove data sources.

Reply:
> the corresponding ticket

Ah, for reference: https://jira.inca.infoblox.com/browse/NGP-6635